scriptparser.py

```diff
@@ -20,61 +20,66 @@ class RemoteScriptBuilder:
     def __init__(self) -> None:
         ...
         
-        
     def get_script_from_user(self, name: str) -> str:
         if name[-3:] != ".py":
             raise Exception("Invalid File Format (Must be '.py'!)")
         self._dir = glob.glob(fr"{pathlib.Path.home().drive}\\**\\{name}")
-        for self.files in self._dir:
+        for self.files in self._dir: # iterating through every file in the directory
             print(f"Combing ('{self.files}')")
             os.system("clear")
             os.system("cls")
-            if self.files == name:
+            if self.files == name: # if we find the target file, we should read the and return the contents
                 with open(name, "r", "utf-8") as file:
                     self.contents = file.read()
                     if self.contents == "":
-                        print(f"Error: Unable to open {name}")
-                        time.sleep(3)
-                        sys.exit()
+                        print(f"Error: {file} was empty..") 
+                        time.sleep(2)
+                        exit(1)
                     return self.contents
         else:
             raise Exception(f"Couldn't find {name} on this device..")
             
             
-    def add_command_to_main_file(self, command_name, script_name, function_name):
+    def add_command_to_main_file(self, command_name: str, script_name: str, function_name: str) -> None:
         with open(".\\boxpyshell.py", "r") as file:
             _contents = file.read()
-        new_contents = f"import {script_name}\n{_contents}\nelif command == '{command_name}':\n    {function_name}({*args}, {**kwargs})"
+        new_contents = f"import {script_name}\n{_contents}\nelif command == '{command_name}':\n    {function_name}({*args}, {**kwargs})" # tbh idk what this does, but it works.. I think
         with open(".\\boxpyshell.py", "w") as file:
             file.write(new_contents)
         
      
-    def add_file_to_list(self, name) -> None:
+    def add_file_to_list(self, name: str) -> None:
         with open("_scripts.txt", "w") as file:
-            if os.path.exists(name) is True and name[-3:] == ".py":
-                file.write(f"{name}")
+            if os.path.exists(name) and name[-3:] == ".py": # if the file exists and the file is a python file, we append the filename to "_scripts.txt"
+                file.write(name) 
                 print(f"Successfully added {name} to '_scripts.txt'")
             else:
                 print(f"{name} is an invalid file (Maybe you forgot to add '.py' to the end?)")
+                exit(1)
         
         
     def build_file(self, name: str) -> None:
-        self._contents = self.get_script_from_user(input("Path to the Script file -> "))
-        self.name = f"{name}.py" if name[-3:] != ".py" else name
+        self._contents: str = self.get_script_from_user(input("Path to the Script file -> "))
```

Comment:
If you want to set the type then instead of doing ```
var: str = ...
```
Rather do
```
var = str(input()) 
```

Reply:
yes